smartsim/_core/launcher/dragon/dragonConnector.py

```diff
@@ -245,9 +245,14 @@ def load_persisted_env(self) -> t.Dict[str, str]:
 
         with open(config.dragon_dotenv, encoding="utf-8") as dot_env:
             for kvp in dot_env.readlines():
-                split = kvp.strip().split("=", maxsplit=1)
-                key, value = split[0], split[-1]
-                self._env_vars[key] = value
+                if not kvp:
```

Comment:
Having trouble leaving comments on unchanged parts for this file:
1. DragonConnector._reset_timeout missing docstring
2. DragonConnector._handshake missing docstring
3. DragonConnector._get_dragon_log_level missing docstring
4. DragonConnector._connect_to_existing_server missing docstring -> dont forget the raises!
5. DragonConnector._start_connector_socket missing docstring - make sure to add the raises
6. DragonConnector._parse_launched_dragon_server_info_from_iterable missing docstring
7. Docstring missing for DragonConnector._parse_launched_dragon_server_info_from_files
8. Docstring missing for DragonConnector._send_req_with_socket
9. Docstring missing for DragonConnector._assert_schema_type
10. DragonConnector._resolve_dragon_path missing docstring